src/FbTk/TextUtils.cc

```diff
@@ -54,8 +54,8 @@ void maxTextLength(int n_pixels, const FbTk::Font& font, const char* const text,
         //             +------left
         //              n_pixels
 
-        int right = n_pixels / (font.textWidth(".", 1) + 1);
-        int left = n_pixels / (font.textWidth("WW", 2) + 1);
+        int right = std::min(n_pixels / (font.textWidth(".", 1) + 1), text_len);
```

Comment:
i do not understand how this solves the off-by-one issue

Reply:
The overrun is in `FbTk::XftFontImp::textWidth(char const*, unsigned int) const (XftFontImp.cc:175)`. This function receives a length that is too long for the text buffer. The change here is a couple of frames up the call stack.

Reply:
the logic of the proposed change here is still wrong, imho: 

`right` is the text-width if the text would consist only of very small glyphs ('.'). the assumption is: more tiny glyphs fit into the available space. to limit `right` to the smaller of either "width of tiny glyphs" or the actual length of the text to be cut is an optimization.

`left`: here we used "wide" glyphs. there will be less wide glyphs which fit into the given space. to limit `left` now to the smaller value of either "wide glyphs" or the actual length of the text to be cut ... is pointless, because then the text would consist out of "WW" in its original form already, it would be wider than what we assume is "wide" text.

neither of the two "trim" down steps will solve the real issue: if the calculated `text_len` at the end is _bigger_ than what is actually there ... that should be limited to `text_len`. so, at the end, `text_lan` must not become bigger than it was in the first place.

if that is the case, then the logic of the algorithm to find a good place where to cut off the text (to make it shorter at rendering) is flawed.

but i do not see how the std::min() approach actually fixes and issue.

Reply:
Maybe you're right. I didn't see the error again, but it could just be a coincidence. I'm not using fluxbox anymore, so do with it what you will. Fluxbox is very responsive while running under valgrind, so its well worth just logging everything for a day or two. Maybe set an alert if any /^==/ appears in the log, so that you can try to guess what caused the problem.